### PR TITLE
Shapes: add arrows and gap markers

### PR DESCRIPTION
**Pull request by danielbeardsley**
Gap markers are for highlighting a linear region or the distance
between two objects.

### CODE REVIEW

File: nudge.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+   /**
+    * Grows the given shape in the given direction by the given number of
+    * pixels relative to its current position.
+    *
+    * @param directionMap A key-value object with the keys {up, down, left,
+    *  right} each with a true/false value.
+    * @param distance The number of pixels to move the shape.
+    */
+   grow: function nudge(directionMap, distance) {
```
Comment (sterlinghirsh) [inline nudge.js]: This looks like nudge is not supposed to be here.

File: two_point_interactivity.js [hunk 2]
```diff
@@ -0,0 +1,266 @@
+var Fabric = require('fabric').fabric || fabric;
+var extend = Fabric.util.object.extend;
+
+module.exports = (function() {
+   var Fabric = require('fabric').fabric || fabric;
```
Comment (sterlinghirsh) [inline two_point_interactivity.js]: Is there any reason to include this here and on line 1?

**Comment by sterlinghirsh** [comment]
CR :zap: But I'm curious about my question about the double require in two_point_interactivity.js.

File: two_point_interactivity.js (hunk 2 above)
Comment (danielbeardsley) [inline two_point_interactivity.js]: Nope, copy pasta mistake.

**Comment by sterlinghirsh** [comment]
CR :zap:

**Pull request merged**